src/LearnMe.Infrastructure/Repository/LessonsRepository.cs

```diff
@@ -0,0 +1,123 @@
+﻿using LearnMe.Infrastructure.Repository.Interfaces;
+using System.Threading.Tasks;
+using LearnMe.Infrastructure.Data;
+using System;
+using LearnMe.Infrastructure.Models.Domains.Lessons;
+using Microsoft.EntityFrameworkCore;
+
+namespace LearnMe.Infrastructure.Repository
+{
+    public class LessonsRepository : CrudRepository<Lesson>, ILessonsRepository
+    {
+        private readonly ApplicationDbContext _context;
+
+        private readonly ICalendarEventsRepository _calendarEventsRepository;
+
+        public LessonsRepository(
+            ApplicationDbContext context,
+            ICalendarEventsRepository calendarEventsRepository) : base(context)
+        {
+            _context = context;
+            _calendarEventsRepository = calendarEventsRepository;
+        }
+
+        public async Task<Lesson> CreateLessonAsync(string calendarId, Lesson lesson)
```

Comment:
do klas repozytorium powinno leciec DTO i tam byc robione mapowanie. Nie mapujemy encji na najwyższej warstwie abstrakcji naszej aplikacji jaką są kontrolery! Macie w `readme.md` napisane, że apka jest pisana w `Clean Architecutre` ale się tutaj tego nie trzymacie. 

Jeśli w Controllerach wstrzykujecie od razu repozytoria i nie macie żadnych command handlerów to na pewno do repo powinno trafić DTO a nie gotowa encja

Reply:
![image](https://user-images.githubusercontent.com/37336963/99898927-bb633880-2ca5-11eb-8235-f5021e015fd2.png)


Chyba nie czujecie czaczy o co chodzi w obiektówce. Po co wam bazowa abstrakcyjna klasa w encji by z niej nie korzystać?

```csharp
ICrudRepository<T> where T : BaseEntity
```

Od biedy jeszcze przeszło by to :

```csharp
public interface ICrudRepository<T> where T : class, new()
```

Ale często encje mają jakieś konstruktory (nie wiem czy u was) także nie za bardzo.

I raczej nie posługujemy się skrótem `obj` w przypadku encji. Parametry lepiej nazwać `T data` albo `T entity`

Reply:
Brak DTO: nie wiem jak to się stało - chyba zapomniałam dodać go do kontrolera, bo mam stworzony mapper w serwisie

czyli 
`public interface ICrudRepository<T> where T : BaseEntity` będzie ok?

Reply:
Będzie ok. Wtedy `CrudRepository` (interfejs) będzie przyjmowało za T w generyku tylko te klasy, które dziedziczą po `BaseEntity` bo wygląda na to, że jest to dla was bazowa klasa dla wszystkich encji także bez przypału.